### DISCUSSION TITLE
how to create prefixed types

### DISCUSSION THREAD

**Discussion by jbbjarnason** [iliekturtles/uom#495] [Q&A]
How can I create prefixed type where the prefix is retained in the quantity

Example:

```rust
        let base_voltage =
            uom::si::i16::ElectricPotential::new::<uom::si::electric_potential::decivolt>(4000);
        assert_eq!(base_voltage.value, 4000);
```

I would like the returned quantity to have the `deci` prefix and the `base_voltage.value` to be `4000`  instead of `400`

**Comment by elrom** [comment]
Some more knowledgeable folks might be able to comment, but afaict this is not supported or intended functionality of UOM. I would like to be wrong, as I too would like this feature.

**Reply by elrom** [reply]
I stand corrected:
https://docs.rs/uom/latest/uom/#design
